### PR TITLE
feature/prepobs.SATEM.fix corrects JDATE and IDSAT types for WCOSS2 compiler

### PR DESCRIPTION
**Pull request by ilianagenkova**
SATEMP was missing from cdas prepbufr files.
SATEMP component of prepbufr is populated by ATOVS dumps.
It was missing due to mismatch in variable types, which the WCOSS2 compiler picked on.
Going forward, watch out for variable type mismatches when *.f90 and *.f code interact.

### CODE REVIEW

**Comment by ilianagenkova** [comment]
@SudhirNadiga-NOAA and @SteveStegall-NOAA , you don't have to review this PR. I tagged you here just so you see what PR progress looks like, and b/c we chatted about this issue yesterday - now you can see what the problem was and how we fixed it. thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
Another lesson learned to highlight ... **Always compile dev code with debug on**!
But when preparing code for hand off for release to NCO, disable debug flags.

File: sorc/prepobs_prepdata.fd/prepdata.f [hunk 1]
```diff
@@ -12672,7 +12672,8 @@ SUBROUTINE SATEDS
       CHARACTER*8   STNPRT,STNID,DSNAME,SUBSET_d
       LOGICAL  SATMST,TR80KM,TOVEDS
       INTEGER  IBUF(140),JRTRV(3),NUMOB(5,4,2,4,4),KEPOB(5,4,2,4),
-     $ MSTOB(5,4,2,4),IDATA(MAXOBS),KSATOB(3),IDSAT(4),ISAMPLE(2)
+     $ MSTOB(5,4,2,4),IDATA(MAXOBS),KSATOB(3),IDSAT(8),ISAMPLE(2) 
+C     MADE IDSAT(4) TO IDSAT(8) to accomodate compiler on WCOSS2
```
Comment (ShelleyMelchior-NOAA) [inline sorc/prepobs_prepdata.fd/prepdata.f]: Put this comment up above around line 12631 in the official documentation timeline. Follow the formatting already in place.

Comment (ShelleyMelchior-NOAA) [reply sorc/prepobs_prepdata.fd/prepdata.f]: Also add comment after line 1337 in main documentation block to indicate this change to the SATEDS routine. Follow the formatting previously established.

Comment (ShelleyMelchior-NOAA) [reply sorc/prepobs_prepdata.fd/prepdata.f]: Also update the W3TAGB parameters in line 3109 to be 2021,0105,0050,'EMC'

File: sorc/prepobs_prepdata.fd/w3xtovseds.f [hunk 2]
```diff
@@ -217,8 +217,8 @@ SUBROUTINE W3XTOVSEDS(IUNIT,IBDATE,IBUFTN,ISATOB,PBOT,DSNAME,
       REAL GEOOPR(18),TLAYER(15),PLAYER(16),RINC(5)
 
       INTEGER IBUFTN(140),ISATOB(3),JDATE(5),JDUMP(5),IBIT(2)
-
-      INTEGER(8) IDSDMP_8,JDUMP_8(5),KDATE(8),LDATE(8)
+C     DEFINED IDSAT as INTEGER(8) to accomodate WCOSS2 compiler
```
Comment (ShelleyMelchior-NOAA) [inline sorc/prepobs_prepdata.fd/w3xtovseds.f]: Put this comment up above around line 70 in the official documentation timeline. Follow the formatting already in place.

File: sorc/prepobs_prepdata.fd/w3xtovseds.f [hunk 3]
```diff
@@ -289,6 +289,7 @@ SUBROUTINE W3XTOVSEDS(IUNIT,IBDATE,IBUFTN,ISATOB,PBOT,DSNAME,
          JFIRST = 1
          IUNITL = IUNIT
          CALL DUMPBF(IUNIT,JDATE,JDUMP)
+         print'(" w3xttovseds output for jdate and jdump ")'
```
Comment (ShelleyMelchior-NOAA) [inline sorc/prepobs_prepdata.fd/w3xtovseds.f]: Remove the second "t" -- should be "w3xtovseds"

**Review by ShelleyMelchior-NOAA — CHANGES_REQUESTED** [review]
Great! So glad you were able to get to the bottom of that very quickly. NCO will be delighted! Please see the documentation and typo corrections. 

Hot tip: any time a code that calls W3TAGB is updated, update the args for that routine.

File: sorc/prepobs_prepdata.fd/prepdata.f (hunk 1 above)
Comment (ShelleyMelchior-NOAA) [reply sorc/prepobs_prepdata.fd/prepdata.f]: My bad ... parameters should be 2022, not 2021

**Comment by ilianagenkova** [comment]
For everyone's learning - in order to compile the code in debug mode, replace in the relevant _build.sh_ :
      -DCMAKE_INSTALL_BINDIR=exec \
with
      -DCMAKE_INSTALL_BINDIR=exec \
      -DCMAKE_BUILD_TYPE=Debug \
i.e. add the Debug option.

File: sorc/prepobs_prepdata.fd/prepdata.f (hunk 1 above)
Comment (ilianagenkova) [reply sorc/prepobs_prepdata.fd/prepdata.f]: Done! I see we use JDay in W3TAGB :)

**Review by ilianagenkova — COMMENTED** [review]
Done, thanks for pointing me to good commenting practices and highlighting W3TAGB!

File: sorc/prepobs_prepdata.fd/prepdata.f (hunk 1 above)
Comment (ShelleyMelchior-NOAA) [reply sorc/prepobs_prepdata.fd/prepdata.f]: Its day of year ... not sure if that is JDay.

**Comment by ShelleyMelchior-NOAA** [comment]
Looks good now!

**Pull request merged**